src/utils/soroban_helper.ts

```diff
@@ -60,57 +69,83 @@ export class SorobanHelper {
     }
   }
 
-  async loadPool(): Promise<Pool> {
-    if (this.pool_cache) {
-      return this.pool_cache;
-    } else {
-      this.pool_cache = await Pool.load(this.network, APP_CONFIG.poolAddress);
-      return this.pool_cache;
+  async loadPool(poolId: string): Promise<Pool> {
+    let cachedPool = this.pool_cache.get(poolId);
+    try {
+      if (cachedPool) {
+        if ('timeout' in cachedPool && cachedPool.timeout > Date.now()) throw cachedPool.error;
+        else if (cachedPool instanceof Pool) return cachedPool;
+      }
+      let pool: Pool = await PoolV1.load(this.network, poolId);
+      this.pool_cache.set(poolId, pool);
+      return pool;
+    } catch (e) {
+      this.pool_cache.set(poolId, { timeout: Date.now() + 5000, error: e });
+      logger.error(`Error loading ${poolId} pool:  ${e}`);
+      throw e;
     }
   }
 
-  async loadUser(address: string): Promise<PoolUser> {
-    if (this.user_cache.has(address)) {
-      return this.user_cache.get(address) as PoolUser;
-    } else {
-      const pool = await this.loadPool();
-      const user = await pool.loadUser(address);
-      this.user_cache.set(address, user);
+  async loadUser(poolId: string, userId: string): Promise<PoolUser> {
+    let cachedUser = this.user_cache.get(poolId + userId);
+    try {
+      if (cachedUser) {
+        if ('timeout' in cachedUser && cachedUser.timeout > Date.now()) throw cachedUser.error;
```

Comment:
we don't need to use a timeout

per usage of this class, it caches everything until you rebuild it.

Most of our workflows are on a per ledger basis, so this is OK

Reply:
this applies for all the fns in this class